src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -152,14 +136,16 @@ std::vector<UChar> GetCustomRules(int32_t options, UColAttributeValue strength,
     bool needsIgnoreWidthCustomRule = isIgnoreWidth && strength >= UCOL_TERTIARY;
     bool needsNotIgnoreWidthCustomRule = !isIgnoreWidth && strength < UCOL_TERTIARY;
 
-    std::vector<UChar> customRules;
+    UCharList* customRules = NULL;
     if (needsIgnoreKanaTypeCustomRule || needsNotIgnoreKanaTypeCustomRule || needsIgnoreWidthCustomRule || needsNotIgnoreWidthCustomRule)
     {
         // If we need to create customRules, the KanaType custom rule will be 88 kana characters * 4 = 352 chars long
         // and the Width custom rule will be at least 215 halfwidth characters * 4 = 860 chars long.
         // Use 512 as the starting size, so the customRules won't have to grow if we are just
         // doing the KanaType custom rule.
-        customRules.reserve(512);
+        customRules = (UCharList*)malloc(sizeof(UCharList*));
+        customRules->items = calloc(512, sizeof(UChar*));
```

Comment:
Is 512 always going to be enough?

Reply:
I have added an AddItem method that checks and reallocate if it goes beyond the capacity.